packages/account/src/Sections/Profile/PhoneVerification/resend-code-timer.tsx

```diff
@@ -1,64 +1,128 @@
 import React from 'react';
 import { Button, CaptionText } from '@deriv-com/quill-ui';
 import { Localize } from '@deriv/translations';
-import { useVerifyEmail } from '@deriv/hooks';
+import { useSettings, useVerifyEmail } from '@deriv/hooks';
+import dayjs from 'dayjs';
 
 type TResendCodeTimer = {
-    resend_code_text: string;
-    count_from: number;
+    timer: number | undefined;
+    is_button_disabled: boolean;
+    should_show_resend_code_button: boolean;
+    setIsButtonDisabled: (value: boolean) => void;
+    setTimer: (value: number | undefined) => void;
     setShouldShowDidntGetTheCodeModal: (value: boolean) => void;
-    start_timer: boolean;
-    setStartTimer: (value: boolean) => void;
+    reInitializeGetSettings: () => void;
 };
 const ResendCodeTimer = ({
-    count_from = 60,
-    resend_code_text,
+    timer,
+    is_button_disabled,
+    should_show_resend_code_button,
+    setIsButtonDisabled,
+    setTimer,
     setShouldShowDidntGetTheCodeModal,
-    start_timer,
-    setStartTimer,
+    reInitializeGetSettings,
 }: TResendCodeTimer) => {
-    // TODO: calculate count_from and time units(secs or mins) using timestamp once mockApi is finalised
-    // TODO: revist start timer logic and localizing the title
-    const [timer, setTimer] = React.useState(count_from);
-    const { send } = useVerifyEmail('phone_number_verification');
-    const initial_timer_title =
-        resend_code_text === 'Resend code' ? `Resend code in ${timer}s` : `Didn’t get the code? (${timer}s)`;
-    const [timer_title, setTimerTitle] = React.useState(initial_timer_title);
+    const [email_otp_attempt, setEmailOtpAttempt] = React.useState('');
```

Comment:
I see here we use either 'email_otp_attempt' or 'phone_otp_attempt', not both of them, right?, so just a suggestion - let's remove two useStates and handle the message with one useState